modules/programs/wezterm.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.wezterm;
+in {
+  options = {
+    programs.wezterm = {
+      enable = mkEnableOption "wezterm";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.wezterm;
+        defaultText = literalExpression "pkgs.wezterm";
+        description = "The Wezterm package to install.";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        example = literalExpression ''
+          local wezterm = require 'wezterm';
+          return {
+            font = wezterm.font("JetBrains Mono"),
+            font_size = 16.0,
+            color_scheme = "Tomorrow Night",
+            hide_tab_bar_if_only_one_tab = true,
+            default_prog = { "zsh", "--login", "-c", "tmux attach -t dev || tmux new -s dev" },
+            keys = {
+              {key="n", mods="SHIFT|CTRL", action="ToggleFullScreen"},
+            }
+          }
+        '';
+        description = ''
+          Configuration written to
+          <filename>$XDG_CONFIG_HOME/wezterm/wezterm.lua</filename>. See
```

Comment:
This breaks config reloading. Please use the workaround mentioned in https://github.com/wez/wezterm/issues/1697

Reply:
Fixed and tested the reloading works.